cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,187 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+
+        self.assertIn(serializer_with_actor1.data, res.data)
+        self.assertIn(serializer_with_actor2.data, res.data)
+        self.assertNotIn(serializer_without_actors.data, res.data)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.genres.add(
+            Genre.objects.create(name="test genre")
+        )
+        movie.actors.add(
+            Actor.objects.create(first_name="Test name", last_name="Test last name")
+        )
+
+        url = detail_url(movie.id)
+        res = self.client.get(url)
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "Test movie",
+            "description": "Test description",
+            "duration": 90,
+        }
+
+        res = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTest(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+        self.admin = get_user_model().objects.create_superuser(
+            email="admin@test.ua",
+            password="testpass"
+        )
+        self.client.force_authenticate(self.admin)
+
+    def test_create_movie(self):
+        payload = {
+            "title": "Test movie",
+            "description": "Test description",
+            "duration": 90,
+        }
+
+        res = self.client.post(MOVIE_URL, payload)
+        movie = Movie.objects.get(id=res.data["id"])
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        for key in payload:
+            self.assertEqual(payload[key], getattr(movie, key))
+
+    def test_create_movie_with_actors_and_genres(self):
+        test_actor1 = sample_actor(first_name="Oleg", last_name="Ol")
+        test_actor2 = sample_actor(first_name="Sava", last_name="Lol")
+
+        test_genre1 = sample_genre(name="horror")
+        test_genre2 = sample_genre(name="Sci-fi")
+
+        payload = {
+            "title": "Test movie",
+            "description": "Test description",
+            "duration": 90,
+            "genres": [test_genre1.id, test_genre2.id],
+            "actors": [test_actor1.id, test_actor2.id]
+        }
+        res = self.client.post(MOVIE_URL, payload)
+        movie = Movie.objects.get(id=res.data["id"])
+        genres = movie.genres.all()
+        actors = movie.actors.all()
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        self.assertIn(test_actor1, actors)
+        self.assertIn(test_actor2, actors)
+        self.assertIn(test_genre1, genres)
+        self.assertIn(test_genre2, genres)
+        self.assertEqual(genres.count(), 2)
+        self.assertEqual(actors.count(), 2)
+
+    def test_delite_movie_not_allowed(self):
```

Comment:
There is a typo in the method name `test_delite_movie_not_allowed`. It should be `test_delete_movie_not_allowed` to correctly reflect the action being tested.